src/core/mount.c

```diff
@@ -1284,6 +1284,11 @@ static int mount_stop(Unit *u) {
 
         assert(m);
 
+        /* When we directly call umount() for a path, then the state of the corresponding mount unit may be
+         * outdated. Let's re-read mountinfo now and update the state. */
+        if (m->dirty)
```

Comment:
i think "dirty" is the wrong term here, that generally indicates data that is not flushed to disk yet. But here we want to mark a state as invalidated.

Hence maybe call this "invalidated"? or "invalidated_state"?

Reply:
Renamed to `invalidated_state`.